File: frontend/src/api/merchant.js
import request from './request'

// ---- 商户：我的店铺 ----
export const getMyShops = (params) => request.get('/merchant/shops', { params })
export const createShop = (data) => request.post('/merchant/shops', data)
export const updateShop = (shopId, data) => request.put(`/merchant/shops/${shopId}`, data)
export const deleteShop = (shopId) => request.delete(`/merchant/shops/${shopId}`)

// ---- 商户：认领未入驻店铺（提交申请，需管理员审核）----
export const getClaimableShops = (params) => request.get('/merchant/shops/claimable', { params })
// 提交认领申请，data: { reason? }；审核通过后店铺才归到自己名下
export const applyClaim = (shopId, data) => request.post(`/merchant/shops/${shopId}/claim`, data)
// 我的认领申请列表（含审核状态与驳回原因）
export const getMyClaims = (params) => request.get('/merchant/claims', { params })
// 撤回待审的认领申请
export const cancelClaim = (claimId) => request.delete(`/merchant/claims/${claimId}`)

// ---- 商户：店铺菜单 ----
export const getShopDishes = (shopId, params) => request.get(`/merchant/shops/${shopId}/dishes`, { params })
export const createDish = (shopId, data) => request.post(`/merchant/shops/${shopId}/dishes`, data)
export const updateDish = (dishId, data) => request.put(`/merchant/dishes/${dishId}`, data)
export const deleteDish = (dishId) => request.delete(`/merchant/dishes/${dishId}`)

File: backend/app/routes/merchant.py
"""商户中心路由（仅商户角色可访问）。"""
from flask import Blueprint, g, request

from app.services.claim_service import ClaimService
from app.services.merchant_service import MerchantService
from app.utils.decorators import require_merchant
from app.utils.responses import ok

bp_merchant = Blueprint('merchant', __name__)


@bp_merchant.route('/shops', methods=['GET'])
@require_merchant
def list_my_shops():
    """我的店铺列表。"""
    return ok(MerchantService.list_my_shops(g.current_user, request.args.to_dict()))


@bp_merchant.route('/shops/claimable', methods=['GET'])
@require_merchant
def list_claimable_shops():
    """可认领的店铺（已公开且未被商户认领，可按 school_id 限定）。"""
    return ok(ClaimService.list_claimable(g.current_user, request.args.to_dict()))


@bp_merchant.route('/shops/<int:shop_id>/claim', methods=['POST'])
@require_merchant
def claim_shop(shop_id):
    """提交认领申请（需管理员审核，通过后才归到自己名下）。"""
    data = request.get_json(silent=True) or {}
    return ok(ClaimService.apply(g.current_user, shop_id, data),
              message='认领申请已提交，等待管理员审核')


@bp_merchant.route('/claims', methods=['GET'])
@require_merchant
def list_my_claims():
    """我的认领申请列表。"""
    return ok(ClaimService.list_mine(g.current_user, request.args.to_dict()))


@bp_merchant.route('/claims/<int:claim_id>', methods=['DELETE'])
@require_merchant
def cancel_claim(claim_id):
    """撤回我的待审认领申请。"""
    ClaimService.cancel(g.current_user, claim_id)
    return ok(None, message='已撤回')


@bp_merchant.route('/shops', methods=['POST'])
@require_merchant
def create_shop():
    """新增我的店铺。"""
    data = request.get_json(silent=True) or {}
    return ok(MerchantService.create_shop(g.current_user, data), message='店铺创建成功')


@bp_merchant.route('/shops/<int:shop_id>', methods=['PUT'])
@require_merchant
def update_shop(shop_id):
    """修改我的店铺。"""
    data = request.get_json(silent=True) or {}
    return ok(MerchantService.update_shop(g.current_user, shop_id, data), message='已更新')


@bp_merchant.route('/shops/<int:shop_id>', methods=['DELETE'])
@require_merchant
def delete_shop(shop_id):
    """删除我的店铺（软删除）。"""
    MerchantService.delete_shop(g.current_user, shop_id)
    return ok(None, message='已删除')


@bp_merchant.route('/shops/<int:shop_id>/dishes', methods=['GET'])
@require_merchant
def list_my_dishes(shop_id):
    """某店铺的菜品列表。"""
    return ok(MerchantService.list_my_dishes(g.current_user, shop_id, request.args.to_dict()))


@bp_merchant.route('/shops/<int:shop_id>/dishes', methods=['POST'])
@require_merchant
def create_dish(shop_id):
    """给店铺添加菜品。"""
    data = request.get_json(silent=True) or {}
    return ok(MerchantService.create_dish(g.current_user, shop_id, data), message='菜品添加成功')


@bp_merchant.route('/dishes/<int:dish_id>', methods=['PUT'])
@require_merchant
def update_dish(dish_id):
    """修改菜品。"""
    data = request.get_json(silent=True) or {}
    return ok(MerchantService.update_dish(g.current_user, dish_id, data), message='已更新')


@bp_merchant.route('/dishes/<int:dish_id>', methods=['DELETE'])
@require_merchant
def delete_dish(dish_id):
    """删除菜品（软删除）。"""
    MerchantService.delete_dish(g.current_user, dish_id)
    return ok(None, message='已删除')
